src/QsFmt/App/Program.fs

```diff
@@ -106,20 +129,174 @@ let runUpdateAndFormat (arguments: UpdateAndFormatArguments) =
+                        (fun (_: IEnumerable<Error>) -> { RunResult.Default with ExitCode = 2 })
+                    )
+                )
+            | ParseArgsResult.Error ex -> Error ( ex )
+        with
+        | ex -> Error ( ex )
+
+    let syntaxErrors =
+        match executionResult with
+        | Success runResult -> Some ( runResult.SyntaxErrors )
+        | Error ex -> None
+
+    let filesProcessed =
+        match executionResult with
+        | Success runResult -> runResult.FilesProcessed
+        | Error ex -> 0
+
+    let exitCode =
+        match executionResult with
+        | Success runResult -> runResult.ExitCode
+        | Error ex -> -1
+
+    let unhandledException =
+        match  executionResult with
+        | Success runResult -> None
+        | Error ex -> Some ( ex )
+
+    let argumentOptions =
+        match parseArgsResult with
+        | ParseArgsResult.Success parsedArgs ->
+            parsedArgs.MapResult(
+                (fun (options: FormatArguments) ->
+                    {
+                        RecurseFlag = options.Recurse
+                        BackupFlag = options.Backup
+                        Command = Format
+                        InputType = if String.IsNullOrEmpty(options.ProjectFile) then Project else Files
+                        QSharpVersion = options.QdkVersion
+                    }
+                ),
+                (fun (options: UpdateArguments) ->
+                    {
+                        RecurseFlag = options.Recurse
+                        BackupFlag = options.Backup
+                        Command = Update
+                        InputType = if String.IsNullOrEmpty(options.ProjectFile) then Project else Files
+                        QSharpVersion = options.QdkVersion
+                    }
+                ),
+                (fun (options: UpdateAndFormatArguments) ->
+                    {
+                        RecurseFlag = options.Recurse
+                        BackupFlag = options.Backup
+                        Command = UpdateFormat
+                        InputType = if String.IsNullOrEmpty(options.ProjectFile) then Project else Files
+                        QSharpVersion = options.QdkVersion
+                    }
+                ),
+                (fun (_: IEnumerable<Error>) -> ArgumentOptions.Default )
+            )
+        | ParseArgsResult.Error ex -> ArgumentOptions.Default
+
+    let executionCompletedEvent =
+        {
+            StartTime = startTime
+            Command = argumentOptions.Command
+            RecurseFlag = argumentOptions.RecurseFlag
+            BackupFlag = argumentOptions.BackupFlag
+            InputType = argumentOptions.InputType
+            QSharpVersion = argumentOptions.QSharpVersion
+            UnhandledException = unhandledException
+            SyntaxErrors = syntaxErrors
+            ExecutionTime = DateTime.Now - startTime
```

Comment:
Since the execution time here is measured on a very small scale, I found that it was highly recommended to use the System.Diagnostics.Stopwatch class to measure the execution time, as it is much more accurate than subtracting DateTime objects. You could still use DateTime.Now to get the start time.

Reply:
Suggestion accepted.